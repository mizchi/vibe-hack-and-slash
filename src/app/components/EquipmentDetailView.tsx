import React, { useState, useEffect } from "react";
import { Box, Text, useInput } from "ink";
import type { Session, Item, EquipmentSlot, Gold } from "../../core/types.ts";
import { processAction } from "../../core/session.ts";
import { getItemDisplayName, getItemStats } from "../../core/loot.ts";
import { calculateTotalStats } from "../../core/combat.ts";
import { getValidSlotsForItem } from "../../core/equipment.ts";
import { calculateStatChanges } from "../utils/stat-preview.ts";
import { calculateItemValue, formatGold } from "../../core/item-value.ts";

type Props = {
  session: Session;
  onSessionUpdate: (session: Session) => void;
  inventory: Item[];
  onInventoryUpdate: (inventory: Item[]) => void;
  battleStatus: {
    isInBattle: boolean;
    currentMonster?: {
      name: string;
      level: number;
      healthPercent: number;
    };
  };
};

const CompactBattleStatus: React.FC<{ battleStatus: Props["battleStatus"] }> = ({ battleStatus }) => {
  if (!battleStatus.isInBattle) {
    return (
      <Box borderStyle="single" padding={1}>
        <Text color="green">◆ 拠点 - 安全 ◆</Text>
      </Box>
    );
  }

  return (
    <Box borderStyle="single" padding={1}>
      <Text color="red">◆ 戦闘中 ◆ </Text>
      {battleStatus.currentMonster && (
        <Text>
          {battleStatus.currentMonster.name} Lv.{battleStatus.currentMonster.level} 
          (HP: {battleStatus.currentMonster.healthPercent}%)
        </Text>
      )}
    </Box>
  );
};

export const EquipmentDetailView: React.FC<Props> = ({
  session,
  onSessionUpdate,
  inventory,
  onInventoryUpdate,
  battleStatus,
}) => {
  const [selectedItemIndex, setSelectedItemIndex] = useState(0);
  const [selectedSlotIndex, setSelectedSlotIndex] = useState(0);
  const [inventoryPage, setInventoryPage] = useState(0);
  const [autoSelectSlot, setAutoSelectSlot] = useState(true);

  const ITEMS_PER_PAGE = 15;
  const playerStats = calculateTotalStats(session.player);

  const totalPages = Math.ceil(inventory.length / ITEMS_PER_PAGE);
  const currentPageItems = inventory.slice(
    inventoryPage * ITEMS_PER_PAGE,
    (inventoryPage + 1) * ITEMS_PER_PAGE
  );

  useInput((input, key) => {
    if (inventory.length > 0) {
      if (key.upArrow) {
        const newIndex = selectedItemIndex - 1;
        if (newIndex < 0) {
          if (inventoryPage > 0) {
            setInventoryPage(inventoryPage - 1);
            setSelectedItemIndex(ITEMS_PER_PAGE - 1);
          }
        } else {
          setSelectedItemIndex(newIndex);
        }
        setAutoSelectSlot(true);
      }
      if (key.downArrow) {
        const newIndex = selectedItemIndex + 1;
        if (newIndex >= currentPageItems.length) {
          if (inventoryPage < totalPages - 1) {
            setInventoryPage(inventoryPage + 1);
            setSelectedItemIndex(0);
          }
        } else {
          setSelectedItemIndex(newIndex);
        }
        setAutoSelectSlot(true);
      }
      if (key.leftArrow || key.rightArrow) {
        setAutoSelectSlot(false);
        setSelectedSlotIndex((prev) => prev + (key.leftArrow ? -1 : 1));
      }
      
      // Space または Enter で装備
      if ((input === " " || key.return) && currentPageItems[selectedItemIndex]) {
        const item = currentPageItems[selectedItemIndex];
        const globalIndex = inventoryPage * ITEMS_PER_PAGE + selectedItemIndex;
        const validSlots = getValidSlotsForItem(item, session.player.class, session.player.level);
        if (validSlots.length > 0) {
          let targetSlot = validSlots[Math.abs(selectedSlotIndex) % validSlots.length];
          if (autoSelectSlot) {
            const itemType = item.baseItem.type;
            if (itemType === "Weapon") {
              targetSlot = "MainHand";
            } else if (itemType === "Armor") {
              if (item.baseItem.tags.includes("Helm")) targetSlot = "Helm";
              else if (item.baseItem.tags.includes("Gloves")) targetSlot = "Gloves";
              else if (item.baseItem.tags.includes("Boots")) targetSlot = "Boots";
              else if (item.baseItem.tags.includes("Belt")) targetSlot = "Belt";
              else targetSlot = "Armor";
            } else if (itemType === "Accessory") {
              if (item.baseItem.tags.includes("Ring")) {
                targetSlot = session.player.equipment.has("Ring1") ? "Ring2" : "Ring1";
              } else if (item.baseItem.tags.includes("Amulet")) {
                targetSlot = "Amulet";
              }
            }
          }
          
          const result = processAction(session, { type: "EquipItem", item, slot: targetSlot });
          if (result.ok) {
            onSessionUpdate(result.value);
            onInventoryUpdate(inventory.filter((_, i) => i !== globalIndex));
            if (currentPageItems.length === 1 && inventoryPage > 0) {
              setInventoryPage(inventoryPage - 1);
              setSelectedItemIndex(0);
            } else {
              setSelectedItemIndex(Math.min(selectedItemIndex, currentPageItems.length - 2));
            }
            setSelectedSlotIndex(0);
            setAutoSelectSlot(true);
          }
        }
      }
      
      // Delete で売却
      if (key.delete && currentPageItems[selectedItemIndex]) {
        const item = currentPageItems[selectedItemIndex];
        const sellValue = calculateItemValue(item);
        const globalIndex = inventoryPage * ITEMS_PER_PAGE + selectedItemIndex;
        
        const updatedSession = {
          ...session,
          player: {
            ...session.player,
            gold: (session.player.gold + sellValue) as Gold,
          }
        };
        onSessionUpdate(updatedSession);
        onInventoryUpdate(inventory.filter((_, i) => i !== globalIndex));
        
        if (currentPageItems.length === 1 && inventoryPage > 0) {
          setInventoryPage(inventoryPage - 1);
          setSelectedItemIndex(0);
        } else {
          setSelectedItemIndex(Math.min(selectedItemIndex, currentPageItems.length - 2));
        }
      }
    }
  });

  return (
    <Box flexDirection="column" height="100%">
      {/* ヘッダー */}
      <Box borderStyle="double" padding={1} marginBottom={1}>
        <Text bold>🎒 装備管理</Text>
        <Text> - </Text>
        <Text color="yellow">{formatGold(session.player.gold)} G</Text>
      </Box>

      {/* 簡易戦闘状況 */}
      <Box marginBottom={1}>
        <CompactBattleStatus battleStatus={battleStatus} />
      </Box>

      <Box flexDirection="row" flexGrow={1}>
        {/* 左側：インベントリ */}
        <Box width="60%" borderStyle="double" padding={1} marginRight={1}>
          <Box>
            <Text bold underline>インベントリ</Text>
            <Text> ({inventory.length}/50) </Text>
            {totalPages > 1 && <Text dimColor>[{inventoryPage + 1}/{totalPages}]</Text>}
          </Box>
          
          {inventory.length === 0 ? (
            <Text dimColor marginTop={1}>アイテムなし</Text>
          ) : (
            <Box flexDirection="column" marginTop={1}>
              {currentPageItems.map((item, index) => {
                const isSelected = index === selectedItemIndex;
                const validSlots = getValidSlotsForItem(item, session.player.class, session.player.level);
                
                const rarityColors = {
                  Common: "gray",
                  Magic: "blue", 
                  Rare: "yellow",
                  Legendary: "magenta",
                };

                return (
                  <Box key={index} flexDirection="column">
                    <Box>
                      <Text color={isSelected ? "cyan" : undefined}>
                        {isSelected ? "▶ " : "  "}
                      </Text>
                      <Text color={rarityColors[item.rarity]}>
                        {getItemDisplayName(item)}
                      </Text>
                    </Box>
                    {isSelected && (
                      <Box marginLeft={3} flexDirection="column">
                        {getItemStats(item).map((stat, i) => (
                          <Text key={i} dimColor>{stat}</Text>
                        ))}
                        {validSlots.length > 0 && (
                          <Text color="green">
                            装備可能: {autoSelectSlot ? "自動選択" : validSlots[Math.abs(selectedSlotIndex) % validSlots.length]}
                          </Text>
                        )}
                        <Text color="yellow">売却価値: {formatGold(calculateItemValue(item))} G</Text>
                      </Box>
                    )}
                  </Box>
                );
              })}
            </Box>
          )}
        </Box>

        {/* 右側：ステータスと装備 */}
        <Box width="40%" flexDirection="column">
          {/* 現在のステータス */}
          <Box borderStyle="single" padding={1} marginBottom={1}>
            <Text bold underline>ステータス</Text>
            <Box marginTop={1}>
              <Text color="green">⚔ 攻撃力: {playerStats.damage}</Text>
              <Text color="cyan">🛡 防御力: {playerStats.defense}</Text>
              <Text color="red">❤️  体力: {playerStats.maxHealth}</Text>
              <Text color="blue">🔮 魔力: {playerStats.maxMana}</Text>
            </Box>
          </Box>

          {/* 装備変更プレビュー */}
          {(currentPageItems[selectedItemIndex] && 
            getValidSlotsForItem(currentPageItems[selectedItemIndex], session.player.class, session.player.level).length > 0) && (
            <Box borderStyle="round" padding={1} marginBottom={1}>
              <Text bold>装備変更プレビュー</Text>
              {(() => {
                const item = currentPageItems[selectedItemIndex];
                const validSlots = getValidSlotsForItem(item, session.player.class, session.player.level);
                const targetSlot = autoSelectSlot 
                  ? validSlots[0]
                  : validSlots[Math.abs(selectedSlotIndex) % validSlots.length];
                const changes = calculateStatChanges(session.player, item, targetSlot);
                
                const formatChange = (stat: { current: number; new: number; diff: number }) => {
                  if (stat.diff === 0) return null;
                  const color = stat.diff > 0 ? "green" : "red";
                  const sign = stat.diff > 0 ? "+" : "";
                  return <Text color={color}> → {stat.new} ({sign}{stat.diff})</Text>;
                };
                
                return (
                  <Box flexDirection="column" marginTop={1}>
                    <Text dimColor>スロット: {targetSlot}</Text>
                    {changes.attack.diff !== 0 && (
                      <Text>⚔ 攻撃: {changes.attack.current}{formatChange(changes.attack)}</Text>
                    )}
                    {changes.defense.diff !== 0 && (
                      <Text>🛡 防御: {changes.defense.current}{formatChange(changes.defense)}</Text>
                    )}
                    {changes.health.diff !== 0 && (
                      <Text>❤️  HP: {changes.health.current}{formatChange(changes.health)}</Text>
                    )}
                    {changes.mana.diff !== 0 && (
                      <Text>🔮 MP: {changes.mana.current}{formatChange(changes.mana)}</Text>
                    )}
                  </Box>
                );
              })()}
            </Box>
          )}

          {/* 装備中アイテム */}
          <Box borderStyle="single" padding={1} flexGrow={1}>
            <Text bold underline>装備中</Text>
            {session.player.equipment.size === 0 ? (
              <Text dimColor marginTop={1}>装備なし</Text>
            ) : (
              <Box flexDirection="column" marginTop={1}>
                {Array.from(session.player.equipment.entries()).map(([slot, item]) => {
                  const rarityColors = {
                    Common: "gray",
                    Magic: "blue", 
                    Rare: "yellow",
                    Legendary: "magenta",
                  };
                  const shortSlotNames: Record<string, string> = {
                    MainHand: "主手",
                    OffHand: "副手",
                    Armor: "鎧",
                    Helm: "兜",
                    Gloves: "手袋",
                    Boots: "靴",
                    Ring1: "指輪1",
                    Ring2: "指輪2",
                    Amulet: "首飾",
                    Belt: "腰帯"
                  };
                  return (
                    <Text key={slot}>
                      <Text dimColor>{shortSlotNames[slot]}: </Text>
                      <Text color={rarityColors[item.rarity]}>
                        {getItemDisplayName(item).substring(0, 25)}
                      </Text>
                    </Text>
                  );
                })}
              </Box>
            )}
          </Box>
        </Box>
      </Box>

      {/* 操作説明 */}
      <Box marginTop={1}>
        <Text dimColor>
          ↑↓: アイテム選択 | Space/Enter: 装備 | ←→: スロット選択 | Del: 売却 | Tab: 戦闘詳細へ
        </Text>
      </Box>
    </Box>
  );
};