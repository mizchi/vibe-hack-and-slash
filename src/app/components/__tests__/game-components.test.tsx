import React from "react";
import { render } from "ink-testing-library";
import { describe, it, expect, beforeAll } from "vitest";

// ゲームコンポーネントのインポート
import { GamePlayView } from "../GamePlayView";
import { BattleDetailView } from "../BattleDetailView";
import { EquipmentDetailView } from "../EquipmentDetailView";

// モックデータ
import type { Session, BaseItem, ItemId, Skill, Monster } from "../../../core/types";

// テスト環境設定
beforeAll(() => {
  process.stdin.isTTY = true;
  process.stdout.isTTY = true;
});

// モックデータの作成
const createMockSession = (): Session => ({
  id: "test-session" as any,
  player: {
    id: "test-player" as any,
    class: "Warrior",
    level: 5 as any,
    currentHealth: 80 as any,
    currentMana: 30 as any,
    experience: 150 as any,
    gold: 250 as any,
    equipment: new Map(),
    name: "テストプレイヤー",
    baseAttributes: {
      strength: 15 as any,
      intelligence: 5 as any,
      dexterity: 10 as any,
      vitality: 12 as any
    },
    baseStats: {
      maxHealth: 100 as any,
      baseDamage: 10 as any,
      criticalChance: 0.1,
      criticalDamage: 1.5,
      lifeSteal: 0,
      mpRegen: 2 as any,
      vitality: 12 as any,
      elementModifier: {
        Physical: 1,
        Arcane: 0,
        Fire: 0,
        Lightning: 0,
        Holy: 0
      }
    },
    inventory: [],
    skills: [],
    skillCooldowns: new Map(),
    skillTimers: new Map(),
    activeBuffs: [],
    elementResistance: {
      Physical: 0,
      Arcane: 0,
      Fire: 0,
      Lightning: 0,
      Holy: 0
    },
    resourcePool: {
      White: 3,
      Red: 0,
      Blue: 0,
      Green: 0,
      Black: 0
    },
    totalResourceLimit: {
      White: 10,
      Red: 10,
      Blue: 10,
      Green: 10,
      Black: 10
    }
  },
  defeatedCount: 3 as any,
  currentMonster: {
    id: "monster-1" as any,
    name: "ゴブリン",
    level: 3 as any,
    currentHealth: 30 as any,
    stats: {
      maxHealth: 50 as any,
      maxMana: 0 as any,
      baseDamage: 10 as any,
      criticalChance: 0.05,
      dodgeChance: 0,
      blockChance: 0,
      blockReduction: 0,
      mpRegen: 0 as any,
      lifeSteal: 0,
      elementModifier: {
        Physical: 1,
        Arcane: 0,
        Fire: 0,
        Lightning: 0,
        Holy: 0
      }
    },
    elementResistance: {
      Physical: 0,
      Arcane: 0,
      Fire: -0.2,
      Lightning: 0,
      Holy: -0.1
    },
    tier: "Common" as const,
    lootTable: []
  },
  state: "InProgress" as const,
  turn: 5,
  droppedItems: [],
  wave: 1 as any
});

const createMockBaseItems = (): Map<ItemId, BaseItem> => {
  const map = new Map<ItemId, BaseItem>();
  map.set("sword-1" as any, {
    id: "sword-1" as any,
    name: "アイアンソード",
    type: "Weapon",
    tags: ["Sword"],
    requiredLevel: 1 as any,
    requiredClass: undefined,
    baseModifiers: [
      {
        type: "IncreaseStrength",
        value: 10
      }
    ],
    weaponType: "Sword",
    weaponUniqueSkills: [],
    weaponScaling: {
      strength: 1,
      intelligence: 0,
      dexterity: 0.5
    }
  });
  return map;
};

// Monster用のダミーデータは不要（GamePlayViewがmonsterTemplatesの型を期待しているため）
const createMockMonsterTemplates = (): any[] => [
  {
    id: "goblin",
    name: "ゴブリン",
    type: "Normal",
    baseStats: {
      maxHealth: 50,
      damage: 10,
      defense: 5,
      speed: 1.0,
      criticalChance: 0.05
    },
    element: "Physical",
    elementResistance: {
      Physical: 0,
      Arcane: 0,
      Fire: -0.2,
      Lightning: 0,
      Holy: -0.1
    },
    xpReward: 10,
    goldReward: 5,
    itemDropRate: 0.2,
    possibleDrops: ["sword-1"]
  }
];

const createMockSkills = (): Skill[] => [
  {
    id: "skill-1" as any,
    name: "パワーストライク",
    description: "強力な一撃",
    type: "Active",
    manaCost: 10 as any,
    cooldown: 2,
    targetType: "Enemy",
    effects: [
      {
        type: "Damage",
        baseDamage: 50 as any,
        scaling: 1.5,
        element: "Physical"
      }
    ],
    requiredLevel: 1 as any,
    weaponType: "Sword"
  }
];

describe("GamePlayView", () => {
  it.skip("ゲームプレイ画面の基本表示", () => {
    const mockSession = createMockSession();
    const mockBaseItems = createMockBaseItems();
    const mockMonsterTemplates = createMockMonsterTemplates();
    const mockSkills = createMockSkills();

    const { lastFrame } = render(
      <GamePlayView
        session={mockSession}
        onSessionUpdate={() => {}}
        baseItems={mockBaseItems}
        monsterTemplates={mockMonsterTemplates}
        skills={mockSkills}
      />
    );

    const output = lastFrame();
    
    // ヘッダー情報
    expect(output).toContain("Hack & Slash");
    expect(output).toContain("rior");  // Warriorが文字化けしている
    expect(output).toContain("Lv.5");
    expect(output).toContain("250");
    
    // バトル情報
    expect(output).toContain("ゴブリン");
    expect(output).toContain("HP:");
    expect(output).toContain("MP:");
  });

  it("スナップショット - ゲームプレイ画面", () => {
    const mockSession = createMockSession();
    const mockBaseItems = createMockBaseItems();
    const mockMonsterTemplates = createMockMonsterTemplates();
    const mockSkills = createMockSkills();

    const { lastFrame } = render(
      <GamePlayView
        session={mockSession}
        onSessionUpdate={() => {}}
        baseItems={mockBaseItems}
        monsterTemplates={mockMonsterTemplates}
        skills={mockSkills}
      />
    );

    expect(lastFrame()).toMatchSnapshot();
  });
});

describe("BattleDetailView", () => {
  it("バトル詳細画面の基本表示", () => {
    const mockSession = createMockSession();
    const battleEvents = [
      { type: "PlayerAttack", damage: 20, isCritical: false },
      { type: "MonsterAttack", damage: 15 },
      { type: "PlayerLevelUp", newLevel: 6 }
    ] as any;

    const { lastFrame } = render(
      <BattleDetailView
        session={mockSession}
        battleLog={battleEvents}
        isPaused={false}
      />
    );

    const output = lastFrame();
    
    // バトル詳細画面の要素
    expect(output).toContain("味方");
    expect(output).toContain("敵");
    expect(output).toContain("戦闘ログ");
  });

  it("スナップショット - バトル詳細画面", () => {
    const mockSession = createMockSession();
    const battleEvents = [
      { type: "PlayerAttack", damage: 20, isCritical: false },
      { type: "MonsterAttack", damage: 15 }
    ] as any;

    const { lastFrame } = render(
      <BattleDetailView
        session={mockSession}
        battleLog={battleEvents}
        isPaused={false}
      />
    );

    expect(lastFrame()).toMatchSnapshot();
  });
});

describe("EquipmentDetailView", () => {
  it.skip("装備詳細画面の基本表示", () => {
    const mockSession = createMockSession();
    const droppedItems = [
      {
        id: "reward-sword" as any,
        baseItem: createMockBaseItems().get("sword-1" as any)!,
        rarity: "Common" as const,
        level: 5 as any,
        modifiers: [
          {
            type: "IncreaseStrength",
            value: 5
          }
        ]
      }
    ];

    const mockBaseItems = createMockBaseItems();
    const mockSkills = createMockSkills();

    const { lastFrame } = render(
      <EquipmentDetailView
        session={mockSession}
        onSessionUpdate={() => {}}
        inventory={droppedItems}
        onInventoryUpdate={() => {}}
        battleStatus={{
          isInBattle: false
        }}
      />
    );

    const output = lastFrame();
    
    // ヘッダー
    expect(output).toContain("インベントリ");
    expect(output).toContain("アイテム詳細");
    
    // アイテム情報
    expect(output).toContain("アイアンソード");
  });

  it("装備なしの場合", () => {
    const mockSession = createMockSession();
    const droppedItems: any[] = [];

    const mockBaseItems = createMockBaseItems();
    const mockSkills = createMockSkills();

    const { lastFrame } = render(
      <EquipmentDetailView
        session={mockSession}
        onSessionUpdate={() => {}}
        inventory={droppedItems}
        onInventoryUpdate={() => {}}
        battleStatus={{
          isInBattle: false
        }}
      />
    );

    const output = lastFrame();
    expect(output).toContain("インベントリ");
    expect(output).toContain("装備可能なアイテムがありません");
  });
});

