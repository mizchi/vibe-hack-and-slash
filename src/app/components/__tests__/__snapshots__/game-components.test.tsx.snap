// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BattleDetailView > スナップショット - バトル詳細画面 1`] = `
"┌────────────────────────────────────────────────┐ ┌───────────────────────────────────────────────┐
│                                                │ │                                               │
│ 味方                                           │ │ 敵 (Wave                                      │
│     Warrior Lv.5                               │ │ 1)       ゴブリン Lv.3                        │
│     HP:███████████░░░░░░░░░░░░░░░░ 80/200 (40%)│ │          HP:██████████████░░░░░░░░░░ 30/50    │
│        █          ░                            │ │             ███           ░         (60%)     │
│     ○3                                         │ │                                               │
│                                                │ │                                               │
└────────────────────────────────────────────────┘ └───────────────────────────────────────────────┘

                                         ┌─────────────────────────────────────────────────────────┐
                                         │                                                         │
                                         │ 戦闘ログ (最新15件)                                     │
                                         │                                                         │
                                         │ 全て | 戦闘 | アイテム | イベント                       │
                                         │                                                         │
                                         │ ▶  [被弾] 15ダメージを受けた                            │
                                         │   [攻撃] 20ダメージを与えた                             │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         │                                                         │
                                         └─────────────────────────────────────────────────────────┘

↑↓: ログ選択 | ←→: フィルター | Enter: 詳細 | P: 一時停止 | R: 休憩"
`;

exports[`GamePlayView > スナップショット - ゲームプレイ画面 1`] = `
"╔══════════════════════════════════════════════════════════════════════════════════════════════════╗
║                                                                                                  ║
║ ⚔️  Hack & Slash - 拠点                                                                          ║
║                                                                                                  ║
╚══════════════════════════════════════════════════════════════════════════════════════════════════╝

┌──────────────────────────────────────┐ ┌─────────────────────────────────────────────────────────┐
│                                      │ │                                                         │
│ ◆ WarH███░░░░   M████░░░ 3EXP:撃破💰  │ │ ◆ メニュー ◆                                            │
│ rior  ███░░░░80/ ████░░░/5150/:  250 │ │                                                         │
│ Lv.5     ░░  200 ██      (00  3体 G  │ │ ▶ 戦闘開始                                              │
│ ◆             (4        0%)          │ │   休憩（全回復）                                        │
│              0%)                     │ │                                                         │
└──────────────────────────────────────┘ └─────────────────────────────────────────────────────────┘

┌──────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                  │
│ バトルログ                                                                                       │
│                                                                                                  │
│                                                                                                  │
└──────────────────────────────────────────────────────────────────────────────────────────────────┘

┌────────────────────────────────────────────────┐ ┌───────────────────────────────────────────────┐
│                                                │ │                                               │
│ インベントリ (0/50)                            │ │ ステー                                        │
│                     アイテムなし               │ │ タス  ⚔ 基礎ダメ🛡    ❤️  体力🔮    ━━    な  │
│                                                │ │       ージ: 10  VIT:   200    魔力: 装備中し  │
│                                                │ │                 12            50    ━━        │
│                                                │ │                                               │
└────────────────────────────────────────────────┘ └───────────────────────────────────────────────┘

Tab: インベントリ | ↑↓: 選択 | Enter: 決定 | Ctrl+C: 終了"
`;
